### ISSUE TITLE
Implement Iterator for Poll

### ISSUE THREAD

**Issue by cjwcommuny** [tokio-rs/mio#1621]
A typical use case of `Poll` is:

```rust
let mut poll = Poll::new()?;
let mut events = Events::with_capacity(128);

// ...

loop {
    poll.poll(&mut events, timeout)?;
    
    for event in events.iter() {
        match event.token() {
            token1 => { ... }
            token2 => { ... }
        }
    }
}
```

It would be better to implement `Iterator<item=io::Result<&Event>>` and `Iterator<item=io::Result<&Events>>` for `Poll`.

The implementation may require GAT, which is expected to be stabilized soon.

**Comment by Thomasdezeeuw** [comment]
I don't think it's a good idea to hide a system call behind an `Iterator` implementation. Furthermore how would you pass the timeout? Also once you get away from the most basic example the `Iterator` implementation wouldn't be useful any more...

**Comment by Noah-Kennedy** [comment]
Yeah, I don't really think that this is a good idea, largely for the reasons Thomas mentioned above.

**Comment by Thomasdezeeuw** [comment]
@cjwcommuny I'm going to close this for now, if you still feel strongly about this we can reopen.
